awslabs/aws-lambda-powertools-typescript
# tests(logger): revisited integration tests

## Description of your changes

This PR refactors the integration tests for Logger and once merged resolves #1330.

The main changes done are:
- All tests now have 3+ invocations and assertions are done on each one of them, rather than only on the first 1-2
- All tests follow the same main structure & conventions
- Fixed a test in the child logger fixture that was not checking the values correctly and was always giving false positives

### How to verify this change

See results of successful run [here](https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/4284657351).

### Related issues, RFCs



**Issue number:** #1330

## Checklist


### Breaking change checklist

***Is it a breaking change?:*** NO


---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
